crates/task-impls/src/quorum_proposal/mod.rs

```diff
@@ -265,23 +242,15 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
         };
 
         // We have three cases to consider:
-        let mut secondary_deps = vec![
+        let secondary_deps = vec![
+            // 1. A QcFormed event
+            AndDependency::from_deps(vec![qc_dependency]),
             // 2. A timeout cert was received
             AndDependency::from_deps(vec![timeout_dependency]),
             // 3. A view sync cert was received.
             AndDependency::from_deps(vec![view_sync_dependency]),
         ];
 
-        // 1. A QcFormed event and QuorumProposalValidated event
-        if *view_number > 1 {
```

Comment:
While I know it's not explicitly required to have received the validated quorum proposal, we do a lot of state updates that *are* used downstream. If we receive a flow of events where the quorum proposal recv was the last one, we'd have already initiated a proposal, potentially proposing before we've have visited the leaf chain, this would break `parent_leaf_an_state`.

Reply:
As discussed on Zulip, I replaced the `QuorumProposalValidated` event handling with `QuorumProposalRecv`.